### PR TITLE
[SPARK-36734][SQL][BUILD][3.1] Upgrade ORC to 1.5.13

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to upgrade Apache ORC from 1.5.12 to 1.5.13 for Apache Spark 3.1.3.

### Why are the changes needed?

Apache ORC 1.5.13 is the latest maintenance release in 1.5.x line having the following patches.
- https://issues.apache.org/jira/projects/ORC/versions/12349322

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
This is Apache ORC 1.5.13 for Apache Spark 3.1.3 which you requested before, @gengliangwang .

File: sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala [hunk 1]
```diff
@@ -693,11 +693,11 @@ class FileBasedDataSourceSuite extends QueryTest
   }
 
   test("SPARK-22790,SPARK-27668: spark.sql.sources.compressionFactor takes effect") {
-    Seq(1.0, 0.5).foreach { compressionFactor =>
+    Seq(1.0, 0.4).foreach { compressionFactor =>
```
Comment (gengliangwang) [inline sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala:696]: why do we need to change this?

**Comment by gengliangwang** [comment]
@dongjoon-hyun Thanks for doing this!

File: sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala (hunk 1 above)
Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala:696]: https://issues.apache.org/jira/browse/ORC-984 starts to add a metadata about a ORC writer version name. That increases a few byte per file.

**Comment by dongjoon-hyun** [comment]
Could you review again, @gengliangwang ? Also, cc @HyukjinKwon

**Comment by dongjoon-hyun** [comment]
@gengliangwang . This one also passed the UTs.

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks, merging to 3.1

**Comment by dongjoon-hyun** [comment]
Thank you!

**Pull request closed without merging**